patterns/observer/observer.js

```diff
@@ -0,0 +1,49 @@
+// Observer pattern defines a one to many dependency relationship from Subject to Observers
+// Observers wait for a signal / trigger from the subject to start.
+// Code reference: https://www.youtube.com/watch?v=PldXoGemkyk
+
+
+// OBSERVER PATTERN WITH CLASS
+
+
+
+class Observable {
+
+    constructor() {
+
+        // The list of subscribers
+        this.subscribers = []
+    }
+
+    // Function is added to the list of subscribers
+    subscribe(fn) {
+        this.subscribers.push(fn);
+    }
+
+    // Function will get removed from the subscribers list
+    unsubscribe(fn) {
+        this.subscribers = this.subscribers.filter((item) => item !== fn);
+    }
+
+    // To broadcast to all the observers
+    broadcast(data) {
+        for (let i = 0; i < this.subscribers.length; i++) {
+            // Calling each function in the observers list with the broadcast data
+            this.subscribers[i](data);
+        }
+    }
+}
+
+
+const observer = new Observable();
```

Comment:
To @Dhuds1 point: This should not be called `observer`, rather this is your subject. Does that make sense? The **subject** is the thing that the **observers** _observe_, so while you are correct in naming your class `Observable`, it might be less awkward to name it `Subject`, then this line would become 

`const subject = new Subject()`

Does that make sense?